libraries/ms-common/src/x86/uart.c

```diff
@@ -1,13 +1,53 @@
 #include "uart.h"
+#include <stdlib.h>
+#include <string.h>
+#include "log.h"
+#include "x86_socket.h"
+
+typedef struct {
+  UARTPort port;
+  X86SocketThread thread;
+  UARTStorage *storage;
+  UARTSettings *settings;
+} SocketData;
+
+static SocketData s_sock[NUM_UART_PORTS];
+
+static void receiver_wrapper(struct X86SocketThread *thread, int client_fd, const char *rx_data,
+                             size_t rx_len, void *context);
+
+void receiver_wrapper(struct X86SocketThread *thread, int client_fd, const char *rx_data,
+                      size_t rx_len, void *context) {
+  // get port number from module_name "uart"+UARTPort
+  UARTPort uart = atoi(thread->module_name + 4) - 1;
+  s_sock[uart].storage->rx_handler((uint8_t *)rx_data, rx_len, context);
```

Comment:
You're passing in the `UARTStorage` as context below, but you're accessing `s_sock` directly and accessing the storage parameter. You're also passing the `UARTStorage` as `context` into the RX handler. Do you understand the purpose of the `context` variable/how it works? I would advise reading the Callback Functions page on Confluence.

Reply:
Not really and thanks for the link. I found the page